src/js/kits/model-kit-select/ModelKitSelect.js

```diff
@@ -0,0 +1,192 @@
+            // update the search param (brand, scale or name)
+            // page is handled separately
+            const { param, value } = action;
+            if (!value) {
+                delete searchParams[param];
+            } else {
+                searchParams[param] = value;
+            }
+
+            // force creation of a new object
+            newState.searchParams = Object.assign({}, searchParams);
+            return newState;
+
+        case 'initial':
+            // include the pre-selected kits that are _still_ selected
+            const { kits } = action;
+            newState.preSelected = kits.filter(kit => selectedIds.includes(kit.id));
+            return newState;
+
+        case 'kitToggle':
+            // remove kits that get unselected, add kits that get selected
+            const { kit, checked } = action;
+            const isPresent = selectedIds.includes(kit.id);
+
+            if (checked && !isPresent) {
+                newState.selectedIds = selectedIds.concat([kit.id]);
+            } else if (!checked && isPresent) {
+                newState.selectedIds = selectedIds.filter(id => id !== kit.id);
+            }
+
+            // remove the kit from the pre selected kits if it gets untoggled - but only
+            // if there are search params
+            if ( !isEmpty(searchParams) && !checked && preSelectedIds.includes(kit.id) ) {
+                newState.preSelected = preSelected.filter(preSelectedKit => preSelectedKit.id !== kit.id);
+            }
+
+            return newState;
+
+        case 'search':
+            // set the search result if it's page one, or append them if it's a higher page.
+            const { results } = action;
+            const newSearchResults = page === 1 ? results : searchResults.concat(results);
+
+            newState.searchResults = newSearchResults;
+            newState.hasNext = results.responseData.next !== null;
+            return newState;
+
+        case 'page':
+            newState.page = action.to;
+            return newState;
+    }
+    throw new Error(`Unknown action: ${action.type}`);
+};
+
+
+const ModelKitSelect = ({ label, htmlName, allowMultiple=false, selected=[] }) => {
+    // track filter parameters & search results
+    const [state, dispatch] = useReducer(
+        reducer,
+        {
+            searchParams: {},
+            page: 1,
+            selectedIds: selected,  // track PKs of kits that are selected
+            preSelected: [],
+            searchResults: [],
+            hasNext: null,
+        }
+    );
+
+    const { searchParams, page } = state;
+
+    // load the preview for selected kit IDs
+    // this is one-off, so no state dependencies!
+    useAsync(() => {
+        const promises = state.selectedIds.map(id => modelKitConsumer.read(id));
+        return Promise
+            .all(promises)
+            .then(kits => {
+                dispatch({
+                    type: 'initial',
```

Comment:
If you wan to have consistent dispatch definitions I'd look into [Flux Standard Action](https://github.com/redux-utilities/flux-standard-action).